### PR TITLE
Vet Clinic : query and update animals table

### PR DESCRIPTION
**Pull request by torobucci**
- [ ] Add a column `species` of type string to animals table.
- [ ] Insert more animal `data` to animals table

Screenshot 1

- [ ] Inside a transaction: 
        -  Update the animals table by setting the `species` column to `unspecified.` 
        -  Verify change was made then roll back the change and verify that the species columns went back to the state before the 
           transaction.
Screenshot 2

- [ ] Inside a transaction:
       - Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
       - Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
       - Verify that changes were made.
       - Commit the transaction.
       - Verify that changes persist after commit.

Screenshot 3

- [ ] Inside a transaction:
       -  delete all records in the animals table, then roll back the transaction.
       - After the rollback verify if all records in the animals table still exists. 
      
Screenshot 4

- [ ] Inside a transaction:
      - Delete all animals born after Jan 1st, 2022.
      - Create a savepoint for the transaction.
      - Update all animals' weight to be their weight multiplied by -1.
      - Rollback to the savepoint
      - Update all animals' weights that are negative to be their weight multiplied by -1.
      - Commit transaction
Screenshot 5
Screenshot 5(2)

- [ ] Write queries to answer the following questions:
      - How many animals are there?
      - How many animals have never tried to escape?
      - What is the average weight of animals?
      - Who escapes the most, neutered or not neutered animals?
           
   
Screenshot 6

      - What is the minimum and maximum weight of each type of animal?
      - What is the average number of escape attempts per animal type of those born between 1990 and 2000?
      
Screenshot 7

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -4,3 +4,19 @@ VALUES
...
+VALUES
+  ('Charmander', '2020-02-08', 0, FALSE, -11),
+  ('Plantmon', '2021-11-15', 2, TRUE, -5.7),
+  ('Squirtle', '1993-04-02', 3, FALSE, -12.13),
+  ('Angemon', '2005-06-12', 1, TRUE, -45),
+  ('Boarmon', '2005-06-07', 7, TRUE, 20.4),
+  ('Blossom', '1998-10-13', 3, TRUE, 17),
+  ('Ditto', '2022-05-14', 4, TRUE, 22);
+
+BEGIN;
```
Comment (Munsa1) [inline data.sql]: - [ ] Kindly place this transaction in `queries.sql` as this is a query and as stated everything that queries the database should be placed in the `queries.sql` file

File: data.sql [hunk 2]
```diff
@@ -4,3 +4,19 @@ VALUES
...
+ALTER TABLE animals
+RENAME COLUMN species TO unspecified;
```
Comment (Munsa1) [inline data.sql]: - [ ] Here we are supposed to set the fields of the `species` column to unspecified not to rename the column itself to unspecified. e.g.

```
UPDATE animals
   SET species = 'unspecified'; 
```
Kindly fix this
image

File: data.sql [hunk 3]
```diff
@@ -4,3 +4,19 @@ VALUES
...
+SELECT * FROM animals;
+ROLLBACK;
```
Comment (Munsa1) [inline data.sql]: - [ ] After `ROLLBACK` you should verify the change was made by adding another `SELECT * FROM animals;`


## _Please note that this whole transaaction has to be moved to `queries.sql`_

File: queries.sql [hunk 4]
```diff
@@ -6,3 +6,42 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+BEGIN;
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+SELECT * FROM animals;
+COMMIT;
+
```
Comment (Munsa1) [inline queries.sql]: - [ ] Again here after the `COMMIT` statement we have to verify the change was made by adding `SELECT * FROM animals;`

File: queries.sql [hunk 5]
```diff
@@ -6,3 +6,42 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+BEGIN;
+DELETE FROM animals
+ROLLBACK;
```
Comment (Munsa1) [inline queries.sql]: - [ ] Again here we have to verify that the change was made using select statements e.g. `SELECT COUNT(*) FROM ANIMALS;`, before and after the `ROLLBACK` statement, kindly fix this.

File: queries.sql [hunk 6]
```diff
@@ -6,3 +6,42 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+SELECT neutered, MAX(escape_attempts) as escape_attempts FROM animals
+ GROUP BY neutered ORDER BY MAX(escape_attempts) DESC;
+SELECT neutered, MAX(weight_kg) as max_weight , MIN(weight_kg) as min_weight FROM animals
+ GROUP BY neutered;
+SELECT species, MAX(weight_kg) as max_weight , MIN(weight_kg) as min_weight FROM animals
+ GROUP BY species;
+SELECT species, AVG(escape_attempts) as avg_escape_attempts FROM animals
+WHERE date_of_birth >= '1990-01-01' AND  date_of_birth < '2000-01-01' GROUP BY species;
+SELECT neutered, AVG(escape_attempts) as avg_escape_attempts FROM animals
+ WHERE date_of_birth >= '1990-01-01' AND  date_of_birth < '2000-01-01' GROUP BY neutered;
```
Comment (Munsa1) [inline queries.sql]: - [ ] Kindly add the query for `What is the average weight of animals?`, you have it in the screenshots you have  provided, but I do not see it here.
Your screenshot
image

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @torobucci, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Added ScreenShots For Queries :heavy_check_mark: 
Descriptive PR :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 3 above)
Comment (torobucci) [reply data.sql]: Made changes
Screenshot 2

File: queries.sql (hunk 5 above)
Comment (torobucci) [reply queries.sql]: Made changes
Screenshot 4

File: queries.sql (hunk 6 above)
Comment (torobucci) [reply queries.sql]: Update queries.sql

**Review by andersonlebon — APPROVED** [review]
## Hi @torobucci 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

nodding-yes

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**